### DISCUSSION TITLE
Coinbase Wallet and Metamask Wallet conflicts

### DISCUSSION THREAD

**Discussion by amit-atr958** [WalletConnect/walletconnect-monorepo#4093] [Developer Support] [answered]
If a user connects to Coinbase wallet then disconnects, and tries to connect the MetaMask wallet, the Coinbase wallet is never disconnected. 
1.Connect Coinbase wallet on a dApp Disconnect wallet 
2.Connect MetaMask wallet to the same dApp Coinbase wallet still remains connected
any solution #2795 #4092 
in walletConnect v3 it is handle but in my project metamask and coinbase and trust wallet use custome button when i click metamask its connect with last connected etherium provider wallet.

**Comment by glitch-txs** [comment, marked as answer]
Use EIP-6963

**Reply by amit-atr958** [reply]
Please share a working example with React and EIP-6963.

**Reply by glitch-txs** [reply]
We have this repo but it's not simplified https://github.com/WalletConnect/EIP6963

this small library can help https://github.com/wevm/mipd
otherwise you have the instructions in the EIP here https://eips.ethereum.org/EIPS/eip-6963

**Reply by amit-atr958** [reply]
Thank you for your support and quick response! I appreciate your assistance.

**Reply by aminpsycho** [reply]
yes
